homeassistant/components/samsungtv/media_player.py

```diff
@@ -348,10 +315,10 @@ def media_previous_track(self):
             await asyncio.sleep(KEY_PRESS_TIMEOUT, self.hass.loop)
         await self.hass.async_add_job(self.send_key, "KEY_ENTER")
 
-    def turn_on(self):
+    async def async_turn_on(self):
         """Turn the media player on."""
-        if self._mac:
-            wakeonlan.send_magic_packet(self._mac, ip_address=self._broadcast)
+        if self._on_script:
+            await self.hass.async_add_job(self._on_script.async_run())
         else:
             self.send_key("KEY_POWERON")
```

Comment:
Schedule this on the executor thread pool with `hass.async_add_executor_job`.

Reply:
This should never be called, already thought about removing. Does HA check the features before calling `turn_off`?

Reply:
Yes. It will check:
https://github.com/home-assistant/home-assistant/blob/0d5486f77201fb71f14025126d6e3cc88c77d4cc/homeassistant/components/media_player/__init__.py#L193-L195

https://github.com/home-assistant/home-assistant/blob/0d5486f77201fb71f14025126d6e3cc88c77d4cc/homeassistant/helpers/service.py#L338-L352

Reply:
So I could remove the `self.send_key("KEY_POWERON")` line.

Reply:
Yes. :+1: